### PR TITLE
[Numpy] Add Documentations

### PR DESCRIPTION
**Pull request by ckt624**
## Description ##
Also add documentations for _np_repeat, _np_ones_like, _npi_expand_dims, _npi_rtrue_divide_scalar, _np_absolute, _np_cbrt, and _np_arccos

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Adds documentations for _np_repeat, _np_ones_like, _npi_expand_dims, _npi_rtrue_divide_scalar, _np_absolute, _np_cbrt, and _np_arccos

## Comments ##
@haojin2

### CODE REVIEW

File: python/mxnet/_numpy_op_doc.py [hunk 1]
```diff
@@ -58,31 +55,53 @@ def _np_reshape(a, newshape, order='C'):
...
-    """Return an array of ones with the same shape and type as a given array.
+    """
+    ones_like(a)
 
+    Return an array of ones with the same shape and type as a given array.
+    
     Parameters
     ----------
-    a : ndarray
+    a : array_like
```
Comment (zoeygxy) [inline python/mxnet/_numpy_op_doc.py]: ``ndarray`` might be better? Does np_ones_like and np_zeros_like support python lists / tuples?

File: python/mxnet/_numpy_op_doc.py [hunk 2]
```diff
@@ -92,37 +111,50 @@ def _np_zeros_like(a):
...
+
+    See Also
+    --------
+    tile : Tile an array.
+
+    Examples
+    --------
+    >>> x = np.array([[1,2],[3,4]])
+    >>> np.repeat(x, 2)
+    array([1, 1, 2, 2, 3, 3, 4, 4])
```
Comment (zoeygxy) [inline python/mxnet/_numpy_op_doc.py]: My program output is of float32 type:
``array([1., 1., 2., 2., 3., 3., 4., 4.])``

File: python/mxnet/ndarray/numpy/_op.py [hunk 3]
```diff
@@ -522,22 +641,45 @@ def swapaxes(a, axis1, axis2):
...
+    """
+    Expand the shape of an array.
     Insert a new axis that will appear at the `axis` position in the expanded
-
+    array shape.
+    
     Parameters
     ----------
-    a : ndarray
+    a : array_like
```
Comment (zoeygxy) [inline python/mxnet/ndarray/numpy/_op.py]: ``ndarray`` might be better

File: python/mxnet/_numpy_op_doc.py (hunk 1 above)
Comment (ckt624) [reply python/mxnet/_numpy_op_doc.py]: Changed. Thx.

File: python/mxnet/_numpy_op_doc.py (hunk 2 above)
Comment (ckt624) [reply python/mxnet/_numpy_op_doc.py]: Changed. Thx.

File: python/mxnet/ndarray/numpy/_op.py (hunk 3 above)
Comment (ckt624) [reply python/mxnet/ndarray/numpy/_op.py]: Changed. Thx.

File: python/mxnet/_numpy_op_doc.py (hunk 1 above)
Comment (zoeygxy) [reply python/mxnet/_numpy_op_doc.py]: @ckt624 Please remember to push your updates.

File: python/mxnet/_numpy_op_doc.py [hunk 4]
```diff
@@ -58,31 +55,53 @@ def _np_reshape(a, newshape, order='C'):
...
+    --------
+    zeros_like : Return an array of zeros with shape and type of input.
+    ones : Return a new array setting values to one.
+    
+    Examples
+    --------
+    >>> x = np.arange(6)
+    >>> x = x.reshape((2, 3))
+    >>> x
+    array([[0, 1, 2],
```
Comment (zoeygxy) [inline python/mxnet/_numpy_op_doc.py]: Mine shows:
```
>>> x
array([[0., 1., 2.],
       [3., 4., 5.]])
```

File: python/mxnet/_numpy_op_doc.py [hunk 5]
```diff
@@ -58,31 +55,53 @@ def _np_reshape(a, newshape, order='C'):
...
+           [3, 4, 5]])
+    >>> np.ones_like(x)
```
Comment (zoeygxy) [inline python/mxnet/_numpy_op_doc.py]: Dtype should also be float.

File: python/mxnet/_numpy_op_doc.py [hunk 6]
```diff
@@ -92,37 +111,50 @@ def _np_zeros_like(a):
...
-    """Repeat elements of an array.
+    """
+    repeat(a, repeats, axis=None)  
+
+    Repeat elements of an array.
 
     Parameters
     ----------
-    a : ndarray
+    a : array_like
```
Comment (zoeygxy) [inline python/mxnet/_numpy_op_doc.py]: should be ndarray

File: python/mxnet/ndarray/numpy/_op.py [hunk 7]
```diff
@@ -28,8 +28,120 @@
...
 
+@set_module('mxnet.ndarray.numpy')
+def absolute(x, out=None, **kwargs):
+    r"""
+    Calculate the absolute value element-wise.
+    np.abs is a shorthand for this function.
+
+    Parameters
+    ----------
+    x : array_like
```
Comment (zoeygxy) [inline python/mxnet/ndarray/numpy/_op.py]: Should be ndarray. Please review and fix other places too.

File: python/mxnet/_numpy_op_doc.py [hunk 8]
```diff
@@ -19,11 +19,10 @@
 
 """Doc placeholder for numpy ops with prefix _np."""
 
-
 def _np_reshape(a, newshape, order='C'):
     """
     reshape(a, newshape, order='C')
-
+
```
Comment (haojin2) [inline python/mxnet/_numpy_op_doc.py]: Why extra whitespace here?

File: python/mxnet/_numpy_op_doc.py [hunk 9]
```diff
@@ -58,31 +55,53 @@ def _np_reshape(a, newshape, order='C'):
...
 
+    Return an array of ones with the same shape and type as a given array.
+    
     Parameters
     ----------
     a : ndarray
         The shape and data-type of `a` define these same attributes of
         the returned array.
-
+
```
Comment (haojin2) [inline python/mxnet/_numpy_op_doc.py]: Why extra whitespaces here?

File: python/mxnet/_numpy_op_doc.py [hunk 10]
```diff
@@ -104,25 +126,35 @@ def _np_repeat(a, repeats, axis=None):
     axis : int, optional
         The axis along which to repeat values.  By default, use the
         flattened input array, and return a flat output array.
-
+
```
Comment (haojin2) [inline python/mxnet/_numpy_op_doc.py]: Please remove all unnecessary whitespaces like these ones.

File: python/mxnet/ndarray/numpy/_op.py [hunk 11]
```diff
@@ -28,8 +28,120 @@
...
+    An ndarray containing the absolute value of each element in x.  
+    
+    Examples
+    ----------
+    >>> x = np.array([-1.2, 1.2])
+    >>> np.absolute(x)
+    array([ 1.2,  1.2])
+    """
+    return _unary_func_helper(x, _npi.absolute, _np.absolute, out=out, **kwargs)
+
```
Comment (haojin2) [inline python/mxnet/ndarray/numpy/_op.py]: 2 blank lines between functions.

Comment (haojin2) [reply python/mxnet/ndarray/numpy/_op.py]: Same for all other similar places

File: python/mxnet/numpy/multiarray.py [hunk 12]
```diff
@@ -46,8 +46,120 @@
...
+    The angle of the ray intersecting the unit circle at the given x-coordinate in radians [0, pi]. 
+    This is a scalar if x is a scalar.
+
+    See also
+    ----------
+    cos, arctan, arcsin
+
+    Notes
+    ----------
+
```
Comment (gyshi) [inline python/mxnet/numpy/multiarray.py]: there are too many whitespace

File: python/mxnet/_numpy_op_doc.py [hunk 13]
```diff
@@ -175,4 +204,4 @@ def _np_cumsum(a, axis=None, dtype=None, out=None):
         result has the same size as `a`, and the same shape as `a` if
         `axis` is not None or `a` is a 1-d array.
     """
-    pass
+    pass
```
Comment (zoeygxy) [inline python/mxnet/_numpy_op_doc.py]: Please add newline at the end of the file.

File: python/mxnet/ndarray/numpy/_op.py [hunk 14]
```diff
@@ -28,8 +28,120 @@
...
+    r"""
+    Calculate the absolute value element-wise.
+    np.abs is a shorthand for this function.
+
+    Parameters
+    ----------
+    x : ndarray
+    Input array.
+
+    out : ndarray, optional
```
Comment (zoeygxy) [inline python/mxnet/ndarray/numpy/_op.py]: Please check whether ``out`` supports tuple as input.

File: python/mxnet/numpy/multiarray.py [hunk 15]
```diff
@@ -1759,22 +1880,45 @@ def swapaxes(a, axis1, axis2):
...
-
+    """
+    Expand the shape of an array.
+    Insert a new axis that will appear at the `axis` position in the expanded
+    array shape.
+    
     Parameters
     ----------
-    a : ndarray
+    a : array_like
```
Comment (zoeygxy) [inline python/mxnet/numpy/multiarray.py]: ndarray

File: python/mxnet/_numpy_op_doc.py (hunk 5 above)
Comment (ckt624) [reply python/mxnet/_numpy_op_doc.py]: Thx. It's float by default.

File: python/mxnet/_numpy_op_doc.py (hunk 9 above)
Comment (ckt624) [reply python/mxnet/_numpy_op_doc.py]: Thx. Fixed.

File: python/mxnet/_numpy_op_doc.py (hunk 8 above)
Comment (ckt624) [reply python/mxnet/_numpy_op_doc.py]: Thx. Fixed.

File: python/mxnet/_numpy_op_doc.py (hunk 10 above)
Comment (ckt624) [reply python/mxnet/_numpy_op_doc.py]: Thx. Fixed.

File: python/mxnet/numpy/multiarray.py [hunk 16]
```diff
@@ -1759,22 +1880,45 @@ def swapaxes(a, axis1, axis2):
 
 @set_module('mxnet.numpy')
 def expand_dims(a, axis):
```
Comment (zoeygxy) [inline python/mxnet/numpy/multiarray.py]: I don't think this function will have the correct signature after rendering. Please build and check, and override the function signature if needed.

File: python/mxnet/ndarray/numpy/_op.py (hunk 14 above)
Comment (ckt624) [reply python/mxnet/ndarray/numpy/_op.py]: Thx and it doesn't.

File: python/mxnet/ndarray/numpy/_op.py (hunk 11 above)
Comment (ckt624) [reply python/mxnet/ndarray/numpy/_op.py]: Thx. Fixed.

File: python/mxnet/symbol/numpy/_symbol.py [hunk 17]
```diff
@@ -31,13 +31,107 @@
...
+    Calculate the absolute value element-wise.
+    np.abs is a shorthand for this function.
+
+    Parameters
+    ----------
+    x : _Symbol
+    Input array.
+
+    out : _Symbol, optional
+    A location into which the result is stored. If provided, it must have a shape
```
Comment (zoeygxy) [inline python/mxnet/symbol/numpy/_symbol.py]: According to the template, all "out" params in _symbol.py had better be commented with
"Dummy parameter to keep the consistency with the ndarray counterpart."

File: python/mxnet/_numpy_op_doc.py (hunk 13 above)
Comment (ckt624) [reply python/mxnet/_numpy_op_doc.py]: Thx. Fixed.

File: python/mxnet/numpy/multiarray.py (hunk 12 above)
Comment (ckt624) [reply python/mxnet/numpy/multiarray.py]: Thx. FIxed.

File: python/mxnet/symbol/numpy/_symbol.py [hunk 18]
```diff
@@ -1006,6 +1100,26 @@ def multiply(x1, x2, out=None):
...
+
+    Parameters
+    ----------
+    x1 : _Symbol
+        Dividend array.
+
+    x2 : _Symbol
+        Divisor array.
+
+    out : ndarray
```
Comment (zoeygxy) [inline python/mxnet/symbol/numpy/_symbol.py]: `out` is also _Symbol

File: python/mxnet/numpy/multiarray.py [hunk 19]
```diff
@@ -1759,22 +1880,45 @@ def swapaxes(a, axis1, axis2):
...
+    reshape : Insert, remove, and combine dimensions, and resize existing ones
+    
+    Examples
+    --------
+    >>> x = np.array([1,2])
+    >>> x.shape
+    (2,)
+    >>> y = np.expand_dims(x, axis=0)
+    >>> y
+    array([[1, 2]])
```
Comment (zoeygxy) [inline python/mxnet/numpy/multiarray.py]: Should be float type

File: python/mxnet/numpy/multiarray.py (hunk 16 above)
Comment (ckt624) [reply python/mxnet/numpy/multiarray.py]: Thx and this signature works.

File: python/mxnet/numpy/multiarray.py (hunk 15 above)
Comment (ckt624) [reply python/mxnet/numpy/multiarray.py]: Thx. Fixed.

File: python/mxnet/symbol/numpy/_symbol.py (hunk 17 above)
Comment (ckt624) [reply python/mxnet/symbol/numpy/_symbol.py]: Thx. Fixed.

File: python/mxnet/symbol/numpy/_symbol.py (hunk 18 above)
Comment (ckt624) [reply python/mxnet/symbol/numpy/_symbol.py]: I just learned from you that it should be "Dummy parameter to keep the consistency with the ndarray counterpart" :)

File: python/mxnet/_numpy_op_doc.py (hunk 4 above)
Comment (ckt624) [reply python/mxnet/_numpy_op_doc.py]: Thx. It's an outdated version.

File: python/mxnet/symbol/numpy/_symbol.py (hunk 18 above)
Comment (zoeygxy) [reply python/mxnet/symbol/numpy/_symbol.py]: This is from the template:
```
out : _Symbol or None
        Dummy parameter to keep the consistency with the ndarray counterpart.
```

File: python/mxnet/numpy/multiarray.py (hunk 19 above)
Comment (ckt624) [reply python/mxnet/numpy/multiarray.py]: Thx. Fixed.

File: python/mxnet/symbol/numpy/_symbol.py (hunk 18 above)
Comment (ckt624) [reply python/mxnet/symbol/numpy/_symbol.py]: Awesome.

**Comment by ckt624** [comment]
I have changed everything from the comments.

File: python/mxnet/numpy/multiarray.py [hunk 20]
```diff
@@ -46,7 +46,122 @@
...
+    ----------	
+    x : ndarray
+    Input array.
+
+    out : ndarray, optional
+    A location into which the result is stored. If provided, it must have a shape 
+    that the inputs broadcast to. If not provided or None, a freshly-allocated array is returned. 
+    A tuple (possible only as a keyword argument) must have length equal to the number of outputs.
+
+    **kwargs
```
Comment (tingying2020) [inline python/mxnet/numpy/multiarray.py]: `**kwargs` should not appear here.

Comment (ckt624) [reply python/mxnet/numpy/multiarray.py]: Fixed. Thx.

File: python/mxnet/symbol/numpy/_symbol.py [hunk 21]
```diff
@@ -31,13 +31,95 @@
...
+@set_module('mxnet.symbol.numpy')
+def absolute(x, out=None, **kwargs):
+    r"""
+    Calculate the absolute value element-wise.
+    np.abs is a shorthand for this function.
+
+    Parameters
+    ----------
+    x : _Symbol
+    Input array.
```
Comment (hgt312) [inline python/mxnet/symbol/numpy/_symbol.py]: indent here
```
x : _Symbol
    Input array.
```

File: python/mxnet/symbol/numpy/_symbol.py [hunk 22]
```diff
@@ -31,13 +31,95 @@
...
+
+    out : _Symbol or None
+    Dummy parameter to keep the consistency with the ndarray counterpart.
```
Comment (hgt312) [inline python/mxnet/symbol/numpy/_symbol.py]: indent here
```
out : _Symbol or None
    Dummy parameter to keep the consistency with the ndarray counterpart.
```

File: python/mxnet/symbol/numpy/_symbol.py [hunk 23]
```diff
@@ -31,13 +31,95 @@
 
 __all__ = ['zeros', 'ones', 'maximum', 'minimum', 'stack', 'concatenate', 'arange', 'argmax',
            'clip', 'add', 'subtract', 'multiply', 'divide', 'mod', 'power', 'split', 'swapaxes',
-           'expand_dims', 'tile', 'linspace', 'sin', 'cos', 'sinh', 'cosh', 'log10', 'sqrt']
+           'expand_dims', 'tile', 'linspace', 'sin', 'cos', 'sinh', 'cosh', 'log10', 'sqrt', 
+           'absolute', 'cbrt', 'arccos']
 
+@set_module('mxnet.symbol.numpy')
```
Comment (hgt312) [inline python/mxnet/symbol/numpy/_symbol.py]: There unary ops' location may need to be at the bottom of this file

File: python/mxnet/symbol/numpy/_symbol.py (hunk 22 above)
Comment (ckt624) [reply python/mxnet/symbol/numpy/_symbol.py]: Changed. Thx.

File: python/mxnet/symbol/numpy/_symbol.py (hunk 21 above)
Comment (ckt624) [reply python/mxnet/symbol/numpy/_symbol.py]: Changed. Thx.

File: python/mxnet/symbol/numpy/_symbol.py (hunk 23 above)
Comment (ckt624) [reply python/mxnet/symbol/numpy/_symbol.py]: Thx. It also works.

File: python/mxnet/_numpy_op_doc.py [hunk 24]
```diff
@@ -19,7 +19,6 @@
 
 """Doc placeholder for numpy ops with prefix _np."""
 
-
```
Comment (haojin2) [inline python/mxnet/_numpy_op_doc.py]: do not remove such lines, we require 2 lines between Python functions.

Comment (ckt624) [reply python/mxnet/_numpy_op_doc.py]: Fixed.

**Pull request closed without merging**